tests/test/org/commcare/util/test/SqlStorageIndexedTests.java

```diff
@@ -0,0 +1,156 @@
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.Vector;
+
+import static junit.framework.Assert.fail;
+import static junit.framework.TestCase.assertFalse;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+public class SqlStorageIndexedTests {
+
+    Case a, b, c, d, e;
+
+    Ledger l, l2, l3;
+
+    SqlIndexedStorageUtility<Case> caseStorage;
+    SqlIndexedStorageUtility<Ledger> ledgerStorage;
+    String owner;
+    String groupOwner;
+    String otherOwner;
+    Vector<String> groupOwned;
+    Vector<String> userOwned;
+
+
+    @Before
+    public void setUp() throws Exception {
+
+        owner = "owner";
+        otherOwner = "otherowner";
+        groupOwner = "groupowned";
+
+        userOwned = new Vector<String>();
+        userOwned.addElement(owner);
+
+        groupOwned = new Vector<String>();
+        groupOwned.addElement(owner);
+        groupOwned.addElement(groupOwner);
+
+        a = new Case("case_name_ipsum", "case_type_ipsum");
+        a.setCaseId("case_id_ipsum");
+        a.setUserId(owner);
+        a.setID(-1);
+
+        l = new Ledger("ledger_entity_id");
+        l.setID(-1);
+        l.setEntry("test_section_id", "test_entry_id", 2345);
+
+        l2 = new Ledger("ledger_entity_id");
+        l2.setID(-1);
+        l2.setEntry("test_section_id_2", "test_entry_id_2", 2345);
+
+        l3 = new Ledger("ledger_entity_id_3");
+        l3.setID(-1);
+        l3.setEntry("test_section_id_3", "test_entry_id_3", 2345);
+    }
+
+    @Test
+    public void testSqlCaseStorage() {
+        try {
+
+            try {
+
+                PrototypeFactory mPrototypeFactory = new PrototypeFactory();
+                mPrototypeFactory.addClass(Case.class);
+
+                String storageKey = "TFCase";
+                String username = "sql-storage-test";
+
+                SqlTestUtils.deleteDatabase(username);
+
+                caseStorage = new SqlIndexedStorageUtility<Case>(Case.class, username, storageKey);
+
+                caseStorage.write(a);
+
+                Case readCase = caseStorage.read(1);
+
+                assertEquals("case_name_ipsum", readCase.getName());
+
+            } catch ( Exception e ) {
+                System.err.println(e.getClass().getName() + ": " + e.getMessage());
+                e.printStackTrace();
+                System.exit(0);
+            }
+
+        } catch (Exception e) {
+            e.printStackTrace();
+            fail("Unexpected exception " + e.getMessage());
+        }
+    }
+
+    @Test
+    public void testSqlLedgerStorage() {
+        try {
+
+            try {
+
+                PrototypeFactory mPrototypeFactory = new PrototypeFactory();
+                mPrototypeFactory.addClass(Ledger.class);
+
+                String storageKey = "Ledger";
+                String username = "wspride";
+
+                SqlTestUtils.deleteDatabase(username);
```

Comment:
==Phillip.

It's worth thinking very hard about how we do this, because if we aren't careful we are going to massively pollute stuff and/or possibly keep links to files that have connectors on them, so deleting the file doesn't do anything, etc. 

Can we

A) make sure the setup/teardown for all of these tests manages the creation of all of these db's
B) Provide some arg somewhere that specifies where the DB files live, that we can point to the builddir, so they'll get cleaned when we clean
C) generally make sure we are sandboxing the sqlite connectors such that we know where the files we are creating go?